src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/CodeActions/Razor/ExtractToCssCodeActionResolver.cs

```diff
@@ -0,0 +1,148 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Buffers;
+using System.IO;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.PooledObjects;
+using Microsoft.CodeAnalysis.Razor.CodeActions.Models;
+using Microsoft.CodeAnalysis.Razor.Formatting;
+using Microsoft.CodeAnalysis.Razor.ProjectSystem;
+using Microsoft.CodeAnalysis.Razor.Protocol;
+using Microsoft.CodeAnalysis.Razor.Utilities;
+using Microsoft.CodeAnalysis.Razor.Workspaces;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Razor.CodeActions;
+
+internal class ExtractToCssCodeActionResolver(
+    LanguageServerFeatureOptions languageServerFeatureOptions,
+    IFileSystem fileSystem) : IRazorCodeActionResolver
+{
+    private readonly LanguageServerFeatureOptions _languageServerFeatureOptions = languageServerFeatureOptions;
+    private readonly IFileSystem _fileSystem = fileSystem;
+
+    public string Action => LanguageServerConstants.CodeActions.ExtractToCssAction;
+
+    public async Task<WorkspaceEdit?> ResolveAsync(DocumentContext documentContext, JsonElement data, RazorFormattingOptions options, CancellationToken cancellationToken)
+    {
+        var actionParams = data.Deserialize<ExtractToCssCodeActionParams>();
+        if (actionParams is null)
+        {
+            return null;
+        }
+
+        if (!documentContext.FileKind.IsComponent())
+        {
+            return null;
+        }
```

Comment:
nit: is this something we can check in Provider to keep all such checks in one place? Seems like by the time we got here all such checks should've been done already.